### PR TITLE
Update clang-format to 16.0.1.

### PR DESCRIPTION
**Pull request by bdice**
This PR updates the clang-format version used by pre-commit.

### CODE REVIEW

File: cpp/scripts/run-clang-format.py
Comment (bdice) [inline cpp/scripts/run-clang-format.py:1]: Can we remove this file? Our style enforcement is centrally handled by pre-commit. I just don't want to delete if some folks happen to rely on it.

**Review by ttnghia — APPROVED** [review]

File: cpp/scripts/run-clang-format.py
Comment (ttnghia) [reply cpp/scripts/run-clang-format.py:1]: I use format-on-save and never touch that file.

**Comment by ttnghia** [comment]
The output of new clang version is much better IMO.

**Review by elstehle — APPROVED** [review]

File: cpp/scripts/run-clang-format.py
Comment (bdice) [reply cpp/scripts/run-clang-format.py:1]: Ok. I will delete it.

**Pull request merged**